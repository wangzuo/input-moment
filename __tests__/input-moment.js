import moment from 'moment';
import React from 'react';
import renderer from 'react-test-renderer';
import InputMoment from '../';

test('render', () => {
  const m = moment().year(2018).month(7).date(8).hours(8).minutes(8).seconds(8);
  const component = <InputMoment moment={m} />;
  expect(renderer.create(component).toJSON()).toMatchSnapshot();
});

test('render with custom labels', () => {
    const m = moment().year(2018).month(7).date(8).hours(8).minutes(8).seconds(8);
    const component = <InputMoment
        moment={m}
        labels={{
            date: 'localisedDate',
            time: 'localisedTime',
            save: 'localisedSave',
            days: {
                sun: 'localisedSun',
                mon: 'localisedMon',
                tue: 'localisedTue',
                wed: 'localisedWed',
                thu: 'localisedThu',
                fri: 'localisedFri',
                sat: 'localisedSat'
            },
            hours: 'localisedHours',
            minutes: 'localisedMinutes'
        }}
    />;
    expect(renderer.create(component).toJSON()).toMatchSnapshot();
});
