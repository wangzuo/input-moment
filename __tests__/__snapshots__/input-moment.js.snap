// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="m-input-moment"
>
  <div
    className="options"
  >
    <button
      className="ion-calendar im-btn is-active"
      onClick={[Function]}
      type="button"
    >
      Date
    </button>
    <button
      className="ion-clock im-btn"
      onClick={[Function]}
      type="button"
    >
      Time
    </button>
  </div>
  <div
    className="tabs"
  >
    <div
      className="m-calendar tab is-active"
    >
      <div
        className="toolbar"
      >
        <button
          className="prev-month"
          onClick={[Function]}
          type="button"
        >
          <i
            className="ion-ios-arrow-left"
          />
        </button>
        <span
          className="current-date"
        >
          August 2018
        </span>
        <button
          className="next-month"
          onClick={[Function]}
          type="button"
        >
          <i
            className="ion-ios-arrow-right"
          />
        </button>
      </div>
      <table>
        <thead>
          <tr>
            <td>
              Sun
            </td>
            <td>
              Mon
            </td>
            <td>
              Tue
            </td>
            <td>
              Wed
            </td>
            <td>
              Thu
            </td>
            <td>
              Fri
            </td>
            <td>
              Sat
            </td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="prev-month"
              onClick={[Function]}
            >
              29
            </td>
            <td
              className="prev-month"
              onClick={[Function]}
            >
              30
            </td>
            <td
              className="prev-month"
              onClick={[Function]}
            >
              31
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              1
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              2
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              3
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              4
            </td>
          </tr>
          <tr>
            <td
              className=""
              onClick={[Function]}
            >
              5
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              6
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              7
            </td>
            <td
              className="current-day"
              onClick={[Function]}
            >
              8
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              9
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              10
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              11
            </td>
          </tr>
          <tr>
            <td
              className=""
              onClick={[Function]}
            >
              12
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              13
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              14
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              15
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              16
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              17
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              18
            </td>
          </tr>
          <tr>
            <td
              className=""
              onClick={[Function]}
            >
              19
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              20
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              21
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              22
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              23
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              24
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              25
            </td>
          </tr>
          <tr>
            <td
              className=""
              onClick={[Function]}
            >
              26
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              27
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              28
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              29
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              30
            </td>
            <td
              className=""
              onClick={[Function]}
            >
              31
            </td>
            <td
              className="next-month"
              onClick={[Function]}
            >
              1
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="m-time tab"
    >
      <div
        className="showtime"
      >
        <span
          className="time"
        >
          08
        </span>
        <span
          className="separater"
        >
          :
        </span>
        <span
          className="time"
        >
          08
        </span>
      </div>
      <div
        className="sliders"
      >
        <div
          className="time-text"
        >
          Hours:
        </div>
        <div
          className="u-slider u-slider-x u-slider-time"
          onClick={[Function]}
        >
          <div
            className="value"
            style={
              Object {
                "width": "34.78260869565217%",
              }
            }
          />
          <div
            className="handle"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "left": "34.78260869565217%",
                "top": "0%",
              }
            }
          />
        </div>
        <div
          className="time-text"
        >
          Minutes:
        </div>
        <div
          className="u-slider u-slider-x u-slider-time"
          onClick={[Function]}
        >
          <div
            className="value"
            style={
              Object {
                "width": "13.559322033898304%",
              }
            }
          />
          <div
            className="handle"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "left": "13.559322033898304%",
                "top": "0%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
